File: Chris Johnson/MTA/JS/MTA.js
console.log("Welcome to the MTA trip planner.")

//attempt 3
const n =["Times Square", "34th", "28th", "23rd", "Union Square", "8th", "9th"];
const l = ["1st", "3rd", "Union Square", "6th", "8th", "55th"]
const six = ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"];

let planTrip = function (entryLine, entryStop, exitLine, exitStop){

// const entryLineConcat = entryLine;
// const entryStopConcat = entryStop;
// const exitLineConcat = exitLine;
// const exitStopConcat = exitStop;

let numberOfStops = 0;
let numberOfStopsAfterExchange = 0;
//console.log("The number of stops after the exchange is " + numberOfStopsAfterExchange);
//console.log(exitStop);

let tripStopsBeforeExchange = "";
let tripStopsAfterExchange = "";
let unionSquarePos = exitLine.indexOf("Union Square");

  if (entryLine !== exitLine){

    numberOfStopsAfterExchange = (exitLine.indexOf("Union Square") - exitLine.indexOf(exitStop));
    if (numberOfStopsAfterExchange < 0){numberOfStopsAfterExchange = numberOfStopsAfterExchange * -1};

    if (exitLine.indexOf(exitStop) > exitLine.indexOf("Union Square") ){
        for (let i = unionSquarePos+1; i < numberOfStopsAfterExchange + unionSquarePos; i++){
        tripStopsAfterExchange = tripStopsAfterExchange + exitLine[i] + ", ";
//console.log("Ran 'if' here");
        }
    }

    else if (exitLine.indexOf(exitStop) < exitLine.indexOf("Union Square") ){
      for (let j = numberOfStopsAfterExchange-1; j >= 0; j--){
      tripStopsAfterExchange = tripStopsAfterExchange + exitLine[j] + ", ";
//      }
    }

//     else {console.log("Error, ran 'if' that shouldn't be run here.")}
//
// console.log(tripStopsAfterExchange);

  exitStop = entryLine.indexOf("Union Square");

// console.log("exitStop is " + exitStop);
}
}

 //if (entryLine === exitLine) {exitStop = entryLine.indexOf(exitStop)};

//determine # stops before exchange
numberOfStopsBeforeExchange = (entryLine.indexOf(entryStop) - exitStop );
if (numberOfStopsBeforeExchange < 0) {numberOfStopsBeforeExchange = numberOfStopsBeforeExchange * -1};

// console.log ("entryLine.indexOf(entryStop)" + " is " + entryLine.indexOf(entryStop) + "exitStop is " + exitStop)
// console.log(numberOfStopsBeforeExchange);

//
// console.log(entryLine.indexOf(entryStop));
// console.log("exitStop is " + entryLine.indexOf(exitStop));


if ( entryLine.indexOf(entryStop) > entryLine.indexOf(exitStop) ){
  for (let l = entryLine.indexOf(entryStop)-1; l >= entryLine.indexOf(entryStop)-1; l--){
    tripStopsBeforeExchange = tripStopsBeforeExchange + entryLine[l] + ", ";
    console.log("Ran 'if' here.");
  }
}


else if  (entryLine.indexOf(entryStop) < entryLine.indexOf(exitStop)){
  for (let k = entryLine.indexOf(entryStop)+1; k < numberOfStopsBeforeExchange + entryLine.indexOf(entryStop); k++){
    tripStopsBeforeExchange = tripStopsBeforeExchange + entryLine[k] + ", ";
  }
console.log("Ran 'else if' here.")
}

console.log("tripStopsAfterExchange is " + tripStopsAfterExchange)  ;
console.log("tripStopsBeforeExchange is " + tripStopsBeforeExchange);
//
// if (entryLine !== exitLine){
console.log ( `Get on the ${entryLine}-line and travel the following  ${numberOfStopsBeforeExchange} stops - ${tripStopsBeforeExchange} until you reach Union Square. Then take the ${exitLine} through the following ${numberOfStopsAfterExchange} stops - ${tripStopsAfterExchange} until you reach your destination of ${exitStop}`);
}



//console.log(tripStopsBeforeExchange)




planTrip(n, "9th", l, "55th");

//
//
//
// /*
// //planTrip('N', 'Times Square', '6', '33rd');
//
// // This is only a suggested function name and signature.
//
// // console.log() shows output similar to this:
// // "You must travel through the following stops on the N line: 34th, 28th, 23rd, Union Square."
// // "Change at Union Square."
// // "Your journey continues through the following stops: 23rd, 28th, 33rd."
// // "7 stops in total."
//
//
// //attempt 2
// const n =["Times Square", "34th", "28th", "23rd", "Union Square", "8th"];
// const l = ["1st", "3rd", "Union Square", "6th", "8th"]
// const six = ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"];
//
//
//
// let planTrip = function (entryLine, entryStop, exitLine, exitStop){
//
//
// let tripStops = "";
// let tripStopsAfterExchange = "";
//
// var exitStopStorage = null;
// let numberOfStopsAfterExchange = null;
//
// //screw you "6" line
// if (entryLine === 6){
//     entryLine = "six"
//   }
// if (exitLine === 6){
//     exitLine = "six"
//   }
//
// // # of stops pre exchange
// let numberOfStops = ( (exitLine.indexOf(exitStop) ) - (entryLine.indexOf(entryStop) ) );
// // incase going backwards
// if (numberOfStops < 0) {numberOfStops = (numberOfStops * -1)};
// //console.log(numberOfStops);
//
//
//
// //First Half
// let travelStart = function (entryLine, entryStop, exitLine, exitStop){
//
// if (entryLine !== exitLine){
// //exitStopStorage = exitLine.indexOf(exitStop);
// exitStop = entryLine.indexOf("Union Square")
// ;}
//
//
// if ( entryLine.indexOf(entryStop) < entryLine.indexOf(exitStop) ) {
//   for (let i = 1; i <= numberOfStops-1; i++){
//     tripStops = tripStops + entryLine[i] + ", "
//   }
// }
// else if ( entryLine.indexOf(entryStop) > entryLine.indexOf(exitStop) ){
//   for (let j = numberOfStops; j >= numberOfStops-1; j--){
//     tripStops = tripStops + entryLine[j] + ", ";
//   }
// }
// console.log(tripStops);
// }
// // travelStart (entryLine, entryStop, exitLine, exitStop)
// travelStart(l, "8th", n, "Times Square");
//
// //work out "numberOfStopsAfterExchange"
// if (entryLine !== exitLine){
//   // # of stops
//
//   numberOfStopsAfterExchange = ( (exitLine.indexOf(exitStop) ) - (exitLine.indexOf("Union Square") ) );
//   // incase going backwards
//   if (numberOfStopsAfterExchange < 0) {numberOfStops = (numberOfStops * -1)};
//
// }}
//
// //console.log(numberOfStopsAfterExchange);
//
// //Second half
// //function definition
// let travelFinish = function (entryLine, entryStop, exitLine, exitStop){
//
// //entryStop = "Union Square";
//
//  //console.log(exitStopStorage);
//  console.log(exitStop);
//
// if ( exitLine.indexOf("Union Square") > exitLine.indexOf(exitStopStorage) ) {
//   //console.log(numberOfStopsAfterExchange-1);
//   for (let i = 0; i <= numberOfStopsAfterExchange-1; i++){
//     tripStopsAfterExchange = tripStopsAfterExchange + exitLine[i] + ", ";
//     //console.log("test");
//   }
// }
//
// else if ( exitLine.indexOf("Union Square") < exitLine.indexOf(exitStopStorage) ){
//   for (let i = numberOfStopsAfterExchange; i > 0; i--){
//     tripStopsAfterExchange = tripStopsAfterExchange + exitLine[i-1] + ", ";
//   //  console.log("test2");
//   }
// }
// console.log(tripStopsAfterExchange)
//
// }
// // if lines don't match run this + second half travel function call
// travelFinish();
//
//
//
//
//
//
//
// planTrip(l, "8th", n, "Times Square");
//
//
//
//
//
//
//
//
//
// //attempt 1
// //
// // let n =["Times Square", "34th", "28th", "23rd", "Union Square", "8th"];
// // let l = ["1st", "3rd", "Union Square", "6th", "8th"]
// // let six = ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"];
// //
// // const lines = {
// //   nNorth : ["Union Square", "23rd", "28th", "34th", "Times Square"],
// //   nSouth: ["Union Square", "8th"],
// //   lNorth: ["Union Square", "3rd", "1st"],
// //   lSouth: ["Union Square", "6th", "8th"],
// //   sixNorth: ["Union Square", "23rd", "28th", "33rd", "Grand Central"],
// //   sixSouth: ["Union Square", "Astor Place"],
// //   };
// //
// //
// //
// //
// // let planTrip = function(entryLine, entryStop, exitLine, exitStop){
// //
// // //northSouthQuery
// // let northSouth = null
// // if (entryLine.indexOf(entryStop) > entryLine.indexOf("Union Square") ) {
// //   northSouth = "South"}
// //   else {northSouth = "North"};
// //
// //
// //
// // }
// //
// //
// //
// // planTrip(l, "1st", l, "8th");
// */

File: Audrey Patricia/week_04/MTA.rb
require 'pry'

def self.logo
  puts"

███╗   ███╗████████╗ █████╗
████╗ ████║╚══██╔══╝██╔══██╗
██╔████╔██║   ██║   ███████║
██║╚██╔╝██║   ██║   ██╔══██║
██║ ╚═╝ ██║   ██║   ██║  ██║
╚═╝     ╚═╝   ╚═╝   ╚═╝  ╚═╝


                  &&&&&&&&&
                &&&
                &&
                &  _____ ___________
                II__|[] | |   I I   |
                |        |_|_  I I  _|
                < OO----OOO   OO---OO
*********************************************************"


end

def set_trip
  self.logo
  puts " "
  puts "Welcome to MTA trip planner"
  puts "type [s] to start"
  puts "type [q] to quit"
end

set_trip
menu_choice = gets.chomp.downcase


$routes = {
  "N" => ["Times Square", "34th", "28th", "23rd", "Union Square", "8th"],
  "L" => ["8th", "6th", "Union Square", "3rd", "1st"],
  "6" => ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"],
}
# sl = starting line, ss = starting stop, el = endling line, es = ending stop
def plan_trip(sl, ss, el, es)
#This is the case for same line travel
  if sl == el
    index_ss = $routes[sl].index(ss)
    index_es = $routes[el].index(es)
    total_stops = (index_es - index_ss).abs
    if index_ss < index_es
      stops = $routes[sl][index_ss..index_es]
    else
      backwards = $routes[sl].reverse()
      index_ss = backwards.index(ss)
      index_es = backwards.index(es)
      stops = backwards[index_ss..index_es]
    end

    stops.shift

    puts "You must travel through the following stops: #{stops} "
    puts "#{total_stops} stops in total"
#This is the case for different line travel
  else
    #First part of the journey
    index_ss = $routes[sl].index(ss)
    index_change1 = $routes[sl].index("Union Square") #union square index from starting line
    total_stops = (index_change1 - index_ss).abs

    if index_ss < index_change1
      stops = $routes[sl][index_ss..index_change1]
    else
      backwards = $routes[sl].reverse()
      index_ss = backwards.index(ss)
      index_change1 = backwards.index("Union Square")
      stops = backwards[index_ss..index_change1]
    end

    stops.shift
    puts "You must travel through the following stops on the #{sl} line: #{stops}"


    puts ">> Change at Union Square"

    index_change2 = $routes[el].index("Union Square") #union square index from ending line
    index_es = $routes[el].index(es)
    total_stops += (index_es - index_change2).abs

    if index_change2 < index_es
      stops = $routes[el][index_change2..index_es]
    else
      backwards = $routes[el].reverse()
      index_es = backwards.index(es)
      index_change2 = backwards.index("Union Square")
      stops = backwards[index_change2..index_es]
    end

    stops.shift

    puts "your journey continues through the following stops on the #{el} line: #{stops}"
    puts ">> #{total_stops} stops in total"
  end

end

until menu_choice == 'q'
  puts `clear`
  case menu_choice

  when 's'
    print "Which line are you on now: "
    sl = gets.chomp.upcase
    print "Which stop are you at on this line: "
    ss = gets.chomp
    print "Which line do you want to go to: "
    el = gets.chomp.upcase
    print "Which stop do you want to get to: "
    es = gets.chomp

    plan_trip(sl, ss, el, es)

  else
    puts "Invalid selection"

  end

  set_trip
  menu_choice = gets.chomp.downcase
end

puts "Thanks for using this trip planner, enjoy your trip"
# plan_trip('N', 'Times Square', '6', 'Grand Central')
